### DISCUSSION TITLE
multiple ipsec site to site, same local and remote peers

### DISCUSSION THREAD

**Discussion by ezeroti** [strongswan/strongswan#2201] [Q&A] [answered]
Hi everyone!

We have infrastructure deployed in AWS, where we have EC2 instances with StronsWan for IPSEC links.
We have the need to be able to set up several IPsec tunnels, using the same peers, both locally and remotely (it is a specific provider that requires a VPN per client).
Is there a module, or some type of configuration to be able to solve this type of architecture without the need to use multiple instances or multiple Wan IPs?

Send example image:
![image](https://github.com/strongswan/strongswan/assets/25537441/ddc3c0e2-bded-4edb-b3cf-bbf60f5b6aa2)

**Comment by tobiasbrunner** [comment]
Are you referring to separate Child/IPsec SAs or IKE SAs? Are the clients and their IP addresses known beforehand?

**Reply by ezeroti** [reply]
Hi! @tobiasbrunner, thanks for response...

Yes, efectively, between client and Providers, peers are know beforehand ( shared information throug spreadsheet or documentation ).
Basicaly, with de same peers, will create multiples/separate IKE SAs, that keep only one server on the client side... its possible this? or i have concepts error?

Thanks...

**Reply by tobiasbrunner** [reply, marked as answer]
If the IPs are known, you can just configure the traffic selectors accordingly in separate child sections of the config. But note that separate IKE_SAs (i.e. connection entries) will require different identities or public IP addresses for each client IP address, otherwise, it's not possible to find the correct config as responder (and uniqueness checks and IKE_SA reuse could also be an issue).

**Reply by ezeroti** [reply]
Thanks Tobias!, finally i'm going for that option Public Ip for each client, so, i have no other choice that modifying the infraestructure.
